internal/store/vpa.go

```diff
@@ -0,0 +1,421 @@
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				ms := []*metric.Metric{}
+				for _, c := range a.Spec.ResourcePolicy.ContainerPolicies {
+					min := c.MinAllowed
+					if mem, ok := min[v1.ResourceMemory]; ok {
+						ms = append(ms, &metric.Metric{
+							LabelKeys:   []string{"container_name"},
+							LabelValues: []string{c.ContainerName},
+							Value:       float64(mem.Value()),
+						})
+					}
+				}
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_vpa_container_resource_policy_max_cpu_cores",
+			Type: metric.Gauge,
+			Help: "Maximum CPU cores the VPA can set for containers matching the name.",
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				ms := []*metric.Metric{}
+				for _, c := range a.Spec.ResourcePolicy.ContainerPolicies {
+					max := c.MaxAllowed
+					if cpu, ok := max[v1.ResourceCPU]; ok {
+						ms = append(ms, &metric.Metric{
+							LabelKeys:   []string{"container_name"},
+							LabelValues: []string{c.ContainerName},
+							Value:       float64(cpu.MilliValue()) / 1000,
+						})
+					}
+				}
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_vpa_container_resource_policy_max_memory_bytes",
+			Type: metric.Gauge,
+			Help: "Maximum memory bytes the VPA can set for containers matching the name.",
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				ms := []*metric.Metric{}
+				for _, c := range a.Spec.ResourcePolicy.ContainerPolicies {
+					max := c.MaxAllowed
+					if mem, ok := max[v1.ResourceMemory]; ok {
+						ms = append(ms, &metric.Metric{
+							LabelKeys:   []string{"container_name"},
+							LabelValues: []string{c.ContainerName},
+							Value:       float64(mem.Value()),
+						})
+					}
+				}
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_vpa_container_status_recommendation_lower_bound_cpu_cores",
+			Type: metric.Gauge,
+			Help: "Minimum CPU cores the container can use before the VPA updater evicts it.",
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				ms := []*metric.Metric{}
+				if a.Status.Recommendation == nil || a.Status.Recommendation.ContainerRecommendations == nil {
+					return &metric.Family{
+						Metrics: ms,
+					}
+				}
+				for _, c := range a.Status.Recommendation.ContainerRecommendations {
+					v := c.LowerBound
+					if cpu, ok := v[v1.ResourceCPU]; ok {
+						ms = append(ms, &metric.Metric{
+							LabelKeys:   []string{"container_name"},
+							LabelValues: []string{c.ContainerName},
+							Value:       float64(cpu.MilliValue()) / 1000,
+						})
+					}
+				}
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_vpa_container_status_recommendation_lower_bound_memory_bytes",
```

Comment:
for resource requests/limits on containers/pods we made this generic, and deprecated the non generic metrics, we should have just the generic one here